libpod/runtime_img.go

```diff
@@ -278,9 +278,16 @@ func (r *Runtime) LoadImage(ctx context.Context, name, inputFile string, writer
 		src       types.ImageReference
 	)
 
+	if name == "" {
+		newImages, err = r.ImageRuntime().LoadAllImagesFromDockerArchive(ctx, inputFile, signaturePolicy, writer)
+		if err == nil {
+			return getImageNames(newImages), nil
+		}
+	}
+
 	for _, referenceFn := range []func() (types.ImageReference, error){
 		func() (types.ImageReference, error) {
-			return dockerarchive.ParseReference(inputFile) // FIXME? We should add dockerarchive.NewReference()
+			return dockerarchive.ParseReference(inputFile)
```

Comment:
So now `name` is completely ignored for `docker-archive:` AFAICS.

Instead, this should do `ParseReference(inputFile+":"+name)`, or even better `ParseNormalizedNamed`+`dockerachive.NewReference` (would reject a "name" `@0`)

Reply:
This was breaking backwards compat (and tests).
`$ podman load -i foo.tar name:tag` is supposed to load the archive and tag the (single) image with `name:tag`.

Reply:
How does that work, when `name` is _not used at all_ now?

Reply:
That's done after in https://github.com/containers/podman/blob/master/pkg/domain/infra/abi/images.go#L465.

It's quite a spaghetti :spaghetti:

Reply:
That uses a _return value_ from this function. I can’t see any way the `name` parameter to this function affects the return value in the docker-archive case.

Reply:
Oops, sorry; it does use `opts.Name` to decide what to tag — but still, this function does not use it to decide what to pull.

Either the `name` parameter indicates what to pull, and in that case it should do so for `docker-archive`, or it doesn’t, and in that case it should be dropped from this function, changing behavior of all the other transports.

Reply:
If we do `input+":"+name` the semantics change. Before, we were loading whatever is inside and add `name` as a tag to it. Now we would expect `name` to be present in the image.

Reply:
Aargh, so this is a Podman invention, not a Docker compatibility CLI.

We have different behavior in different transports, and none of `docs/podman-load.1`, `cmd/podman/images/load.go`, `ImageLoadOptions` actually document what the “name” field means. What a _mess_.

I can’t spend a lot of time on this now; I guess cleaning that up (if it should be cleaned up) is well outside the scope of this PR, but I’d at least like to insist on very explicit documentation  (end-user and internal throughout the API / data structures) of what the expected behavior actually is.

Reply:
I will add a comment and open an issue so we don't forget about it.

Reply:
> 我将添加评论并打开一个问题，这样我们就不会忘记它。

https://github.com/containers/podman/issues/7337 I think it's the same issue